core/transactor/tracker/noncer.go

```diff
@@ -72,9 +74,11 @@ func (n *Noncer) refreshNonces(ctx context.Context) {
 		// TODO: handle case where stored & chain pending nonce is out of sync?
 	}
 
-	if content, err := n.ethClient.TxPoolContent(ctx); err == nil {
-		for _, tx := range content["queued"][n.sender.Hex()] {
-			n.queuedNonces[tx.Nonce()] = struct{}{}
+	// Use txpool.inspect instead of txpool.content. Less data to fetch.
+	if content, err := n.ethClient.TxPoolInspect(ctx); err == nil {
+		for nonceStr := range content["pending"][n.sender] {
+			nonce, _ := strconv.ParseUint(nonceStr, 10, 64)
+			n.pendingNonces[nonce] = struct{}{}
```

Comment:
as per discussion, add both pending and queued nonces (and rename to smth like `inMempoolNonces`) as both pending and queued require a new tx to have bumped gas it to go through